src/EditorFeatures/CSharpTest/SplitOrMergeIfStatements/MergeConsecutiveIfStatementsTests.cs

```diff
@@ -0,0 +1,1685 @@
+        public async Task MergedIntoParentWithDifferenceInBlocks1()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a)
+            System.Console.WriteLine(a || b);
+        else [||]if (b)
+        {
+            System.Console.WriteLine(a || b);
+        }
+    }
+}",
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a || b)
+            System.Console.WriteLine(a || b);
+    }
+}");
+        }
+
+        [Fact]
+        public async Task MergedIntoParentWithDifferenceInBlocks2()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a)
+        {
+            System.Console.WriteLine(a || b);
+        }
+        else [||]if (b)
+            System.Console.WriteLine(a || b);
+    }
+}",
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a || b)
+        {
+            System.Console.WriteLine(a || b);
+        }
+    }
+}");
+        }
+
+        [Fact]
+        public async Task MergedIntoParentWithDifferenceInBlocks3()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a)
+        {
+            System.Console.WriteLine(a || b);
+        }
+        else [||]if (b)
+        {
+            {
+                System.Console.WriteLine(a || b);
+            }
+        }
+    }
+}",
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a || b)
+        {
+            System.Console.WriteLine(a || b);
+        }
+    }
+}");
+        }
+
+        [Fact]
+        public async Task NotMergedIntoParentWithUnmatchingStatements1()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a)
+        {
+            System.Console.WriteLine(a || b);
+        }
+        else [||]if (b)
+        {
+            System.Console.WriteLine(a || a);
+        }
+    }
+}");
+        }
+
+        [Fact]
+        public async Task NotMergedIntoParentWithUnmatchingStatements2()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a)
+            System.Console.WriteLine(a || b);
+        else [||]if (b)
+            System.Console.WriteLine(a || a);
+    }
+}");
+        }
+
+        [Fact]
+        public async Task NotMergedIntoParentWithUnmatchingStatements3()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a)
+        {
+            System.Console.WriteLine(a);
+        }
+        else [||]if (b)
+            System.Console.WriteLine(b);
+    }
+}");
+        }
+
+        [Fact]
+        public async Task NotMergedIntoParentWithUnmatchingStatements4()
+        {
+            // Do not consider the using statement to be a simple block (as might be suggested by some language-agnostic helpers).
```

Comment:
which helpers consider that?  that seems like a bug in them.

Reply:
No, it's not a bug in any helper. I first used `IsStatementContainer` which was a mistake on my part and had this bug. So I added this test as a precaution because it may be tempting to use that or `IsExecutableBlock` without thinking about what that means.